rpc/api.py

```diff
@@ -519,6 +527,165 @@ def get_queryset(self):
         """Get clusters with RFC number annotations"""
         return Cluster.objects.with_rfc_number_annotated()
 
+    @extend_schema(
+        operation_id="clusters_add_document",
+        request=ClusterAddRemoveDocumentSerializer,
+        responses=ClusterSerializer,
+        examples=[
+            OpenApiExample(
+                "Add Document to Cluster",
+                value={"draft_name": "draft-ietf-example-document"},
+                request_only=True,
+            ),
+        ],
+    )
+    @action(detail=True, methods=["post"], url_path="add-document")
+    def add_document(self, request, number=None):
+        """Add a document to a cluster"""
+        cluster = self.get_object()
+
+        serializer = ClusterAddRemoveDocumentSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        draft_name = serializer.validated_data["draft_name"]
+
+        # Get the Document by draft name
+        try:
+            doc = Document.objects.get(name=draft_name)
+        except Document.DoesNotExist:
+            raise serializers.ValidationError(
+                {"draft_name": [f"Document with name '{draft_name}' not found"]},
+                code="document_not_found",
+            ) from None
+
+        # Check if document is already in any cluster
+        existing_member = ClusterMember.objects.filter(doc=doc).first()
+        if existing_member:
+            raise serializers.ValidationError(
+                {
+                    "draft_name": [
+                        f"Document {draft_name} is already in cluster "
+                        f"{existing_member.cluster.number}"
+                    ]
+                },
+                code="document_already_in_cluster",
+            )
+
+        max_order = (
+            ClusterMember.objects.filter(cluster=cluster).aggregate(Max("order"))[
+                "order__max"
+            ]
+            or 1
+        )
+
+        ClusterMember.objects.create(cluster=cluster, doc=doc, order=max_order + 1)
+
+        cluster.refresh_from_db()
+
+        serializer = self.get_serializer(cluster)
+        return Response(serializer.data)
+
+    @extend_schema(
+        operation_id="clusters_remove_document",
+        request=ClusterAddRemoveDocumentSerializer,
```

Comment:
suggestion above about moving the request serializer into the `@action()` applies here as well